Start work on the new custom world border

fooeyround opened the pull request:
This uses polymer entity attachments with block display entities. 

Issues include

- [x] Every player except the player that should see the elements see them.
-  [ ] The Texture / BlockState is currently bedrock, this should become different states of a translucent block (probably stained glass with a custom texture)
- [x] The custom world border's logic / main class is not yet finished, only the renderer has been worked on.
- [x] The border's transformations are not interpolated.
-  [x] The border's transformations should be done from source -> target, allowing for the interpolation to take effect on the client.

(There are other issues not yet mentioned)

fooeyround committed:
start work on custom world border with block displays

fooeyround committed:
create a dummy translation; TODO: fix all of the todos

fooeyround committed:
working interpolation, now using markDirty() like some parts of vanilla

JacktheDevel0per commented:
interpolation is working now... the transformations now just have to be set only when the border state updates. This does also mean that the check for if you are inside or outside of the border need to keep track of this and do a simple lerp.

fooeyround committed:
it is a mess, I was testing some ways to get the postion to be faked.. The translation needs to be kept in 2 layers...

senseiwells committed:
Update the custom WB
- Fix some events to block WB packets
- Make the custom WB use vanilla WB as the back-end (allows for /worldborder)

senseiwells committed:
Fix border being displayed to others
- Added some more notes

fooeyround committed:
remove redundant override. start work on an isDirty check and Interpolation. FIx issue with world border location.

JacktheDevel0per commented:
For the interpolation I am pretty sure we just need to re send a new position, without interpolated movement / reset the position to the "active position" then just send a new movement with the remaining ticks.

The issue I am currently having is being able to correctly reset the lerp and then start it again, without the translation freezing, the scaling can always be lerped / updated only when Broken by the translation reset or the border starting to move. Another problem might be that the to and from in border extent might be needed.
